napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+
+    for i in range(n_channels):
+        if i != ref_channel:
+            # Create table (label_links) that links labels from probe channel
+            # to reference channel
+            label_links = pd.DataFrame(
+                sk_regionprops_table(label_image=labels_array[i],
+                                     intensity_image=labels_array[ref_channel],
+                                     properties=['label', ],
+                                     extra_properties=[highest_overlap]
+                                     )
+            ).astype(int)
+            # rename column
+            label_links.rename(columns={'label': 'label' + suffixes[i],
+                                        'highest_overlap':
+                                            'label' + suffixes[ref_channel]},
+                               inplace=True)
+
+            # Include extra properties of reference channel
+            properties_with_extras = [props for props in ref_channel_props
+                                      if props != 'label']
+            
+            # Add missing reference labels to table (they belong to background)
+            for ref_label in ref_channel_props['label'].tolist():
+                if ref_label not in label_links['label' + suffixes[ref_channel]].values:
+                    label_links.loc[-1] = [0, ref_label] # adding a row
+                    label_links.index = label_links.index + 1 # shifting index
+            label_links = label_links.sort_index() # sorting by index
+            
+            # Append properties of reference channel to table
+            for props in properties_with_extras:
+                large_numbers = False
+                # If large numbers, store format and convert to string to allow
+                # pandas replacement below, restore data type afterwards
+                ref_channel_props_dropped_nans = ref_channel_props[props][
+                    ~np.isnan(ref_channel_props[props])]
+                if abs(ref_channel_props_dropped_nans).max() > np.iinfo(np.int32).max:
+                    large_numbers = True
+                    props_dtype = ref_channel_props[props].dtype
+                    ref_channel_props[props] = \
+                        ref_channel_props[props].astype(str)
+
+                props_mapping = make_element_wise_dict(
+                    ref_channel_props['label'].tolist(),
+                    ref_channel_props[props].tolist())
+
+                label_links[props + suffixes[ref_channel]] = \
+                    label_links['label' + suffixes[ref_channel]]
+
+                label_links = label_links.replace(
+                    {props + suffixes[ref_channel]: props_mapping}
+                    )
+
+                if large_numbers:
+                    label_links[props + suffixes[ref_channel]] = \
+                        label_links[props + suffixes[ref_channel]]\
+                        .astype(props_dtype)
```

Comment:
I don't understand what's happening here. It appears tables are concatenated/merged. I'm wondering if there is functionality in pandas which might make this code easier to read.

Reply:
Please read my comment below regarding `merge`.